python/semantic_kernel/connectors/ai/ollama/services/utils.py

```diff
@@ -115,11 +114,17 @@ def update_settings_from_function_choice_configuration(
     settings: PromptExecutionSettings,
     type: FunctionChoiceType,
 ) -> None:
-    """Update the settings from a FunctionChoiceConfiguration."""
-    assert isinstance(settings, OllamaChatPromptExecutionSettings)  # nosec
+    """Update the settings from a FunctionChoiceConfiguration.
 
+    Since this function might be called before the settings are cast to Ollama Settings
```

Comment:
Is this still true after the latest changes?

Reply:
yeah, this is a loose function that might or might not be called with the right settings so needs to be able to do both, or we need to change the signature to only be OllamaChatPromptExecutionSettings...